algebird-core/src/main/scala/com/twitter/algebird/SummingCache.scala

```diff
@@ -58,7 +58,7 @@ class SummingCache[K, V](capacity: Int)(implicit sgv: Semigroup[V])
   }
   override def flush: Option[Map[K, V]] = {
     // Get a copy of the cache, since it is mutable
-    val res = optNonEmpty(Map(cache.toSeq: _*))
+    val res = optNonEmpty(cache.iterator.toMap)
```

Comment:
I didn't benchmark. I was thinking about using this to implement mapSide caching, which could have allowed at least using sumOption on the "reducer" side of the computation, but I back-peddled on that since we'd have to have parameters for the cachesize, etc...

When I looked at this, I noticed the existing algorithm is going through the cache twice: once to create the Seq. a second time to create the map. Using the iterator, you go through it only one time. Seems hard to imagine how this can hurt, but it probably does not help much as flush does not occur very often.